src/map/script.c

```diff
@@ -12618,13 +12623,49 @@ BUILDIN(waitingroomkickall) {
 	return true;
 }
 
+/// Kicks target player from the waiting room of the current or target npc.
+///
+/// kickwaitingroom "<npc_name>"{,"<name>"|<account id>};
+/// kickwaitingroom;
+BUILDIN(waitingroomkick) {
+	struct npc_data *nd;
+	struct chat_data *cd;
+	struct map_session_data *sd = NULL;
+
+	if (script_hasdata(st, 2)) {
+		nd = npc->name2id(script_getstr(st, 2));
+		if (nd == NULL) {
+			ShowWarning("buildin_waitingroomkick: NPC name '%s' not found.\n", script_getstr(st, 2));
+			return false;
+		}
+	}
+	else
+		nd = map->id2nd(st->oid);
+
+	if (nd != NULL && (cd = map->id2cd(nd->chat_id)) != NULL) {
+		if (script_hasdata(st, 3)) {
+			if (script_isstringtype(st, 3))
+				sd = map->nick2sd(script_getstr(st, 3));
+			else
+				sd = map->id2sd(script_getnum(st, 3));
```

Comment:
don't use `map->nick2sd` or `map->id2sd` here
this is only meant for checking for conditions
in this case since this script command wants to throw error for player not found
use `script->nick2sd` and `script->id2sd`

see https://github.com/HerculesWS/Hercules/pull/950